pallets/parachain-staking/src/weights.rs

```diff
@@ -88,6 +89,12 @@ pub trait WeightInfo {
 /// Weights for parachain_staking using the Substrate node and recommended hardware.
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	fn hotfix_remove_delegation_requests(x: u32) -> Weight {
```

Comment:
This looks good at first glance, although it's unclear how `x` maps to the size of the vec passed in. We should do some manual tests on this to make sure it works as expected...

Reply:
The input to the hotfix is the `delegators: Vec<T::AccountId>`. For each `T::AccountId` in that vec, we are applying the hotfix so the overall cost of executing the extrinsic is `delegators.len() * cost_of_executing_one_hotfix`. The `x` is `delegators.len() as u32` for that reason.

> We should do some manual tests on this to make sure it works as expected...

This is the benchmarking code. I'm pretty sure it is right because the hotfix does one read, one write per T::AccountId passed in as well as changing the state by removing the leftover delegations.

Reply:
Sorry, I meant that it is unclear how Substrate maps x to the Vec length -- the benchmark itself looks good to me.